testcafe/tests/pages/photos-album-model.js

```diff
@@ -0,0 +1,263 @@
+import { t, Selector } from 'testcafe'
+import {
+  getPageUrl,
+  getElementWithTestId,
+  isExistingAndVisibile,
+  checkAllImagesExists
+} from '../helpers/utils'
+import PhotoPage from '../pages/photos-model'
+const photoPage = new PhotoPage()
+
+export default class Page {
+  constructor() {
+    this.albumEmptyText = photoPage.folderEmpty.withText(
+      "You don't have any album yet"
+    ) //!FIXME : text !
+
+    this.albumContentWrapper = getElementWithTestId('album-pho-content-wrapper')
+    this.albumTitle = getElementWithTestId('pho-content-title')
+
+    // Album list
+    this.toolbarAlbumsList = getElementWithTestId('pho-toolbar-albums')
+    this.btnNewAlbum = getElementWithTestId('album-add')
+    this.album = albumName => {
+      return getElementWithTestId('pho-album').withAttribute(
+        'data-test-name',
+        albumName
+      )
+    }
+
+    //New albums & edit album
+    this.inputAlbumName = getElementWithTestId('input-album-name')
+    this.pickerAlbumName = getElementWithTestId('pho-picker-album-name')
+    this.btnValidateAlbum = getElementWithTestId('validate-album') //Seme button for create album and Add to album
+    this.photoThumb = value => {
+      return Selector('[class*="pho-photo-item"]').nth(value)
+    }
+    this.photoSectionAddToAlbum = getElementWithTestId('picker-panel')
+      .find('div')
+      .withAttribute('data-test-id', 'photo-section')
+
+    this.allPhotosAddToAlbum = this.photoSectionAddToAlbum
+      .find('img')
+      .parent('div')
+      .withAttribute('data-test-id', 'pho-photo-item')
```

Comment:
No in this case : 
`data-test-id='pho-photo-item'` is on all photos, but when adding pictures in an album, both photos that already are in the album and photos to add to the album are loaded in the page. 

So I need to pick the images only in ` this.photoSectionAddToAlbum` but i cannot do 
` this.photoSectionAddToAlbum.withAttribute('data-test-id', 'pho-photo-item')` 
This would check if there is a data-test-id in the DOM element `this.photoSectionAddToAlbum`, but not in the child nodes.

I can try 
`this.allPhotosAddToAlbum = this.photoSectionAddToAlbum.find('div').withAttribute(.... `
So it find div with the right attribute inside `this.photoSectionAddToAlbum`without checking the img if you want

Reply:
No that's ok, I was missing the "`data-test-id='pho-photo-item'` is on all photos" context. However I would assume `id` is unique per page, so maybe we should use a different `data-test` attribute when multiple elements in the page have the same one?

Reply:
`data-test-item` or `data-test-el` to identify elements that shows up several time like photo-items, in photos, or rows in drive